src/components/document/canvas.tsx

```diff
@@ -88,8 +87,10 @@ export class CanvasComponent extends BaseComponent<IProps, IState> {
     };
   }
 
-  private getIdClass() {
-    return this.props.idClass ?? "basic";
+  private setCanvasRef(canvasRef?: HTMLDivElement | null) {
+    if (!this.state.canvasRef) {
```

Comment:
I'm pretty sure React takes care of this internally. If you are setting the state to be the same value that is already set, it won't trigger a re-render.

Because of this, you could consider dropping this function and instead down below just do:
`ref={canvasRef => this.setState({canvasRef})}`

Reply:
I originally tried to do what you're suggesting here, but I was getting an infinite render loop. I just double checked and it's still doing the same thing. 🤷